src/crsAdapter/BewotecExpertAdapter.js

```diff
@@ -302,6 +302,10 @@ class BewotecExpertAdapter {
     getDataFromBewotecBridge(options) {
         return new Promise((resolve, reject) => {
             const bewotecDataListener = (message) => {
+                if (!this.options.debug && this.bridgeWindow && !this.bridgeWindow.closed) {
+                    this.bridgeWindow.close();
+                }
```

Comment:
you have to move that more to the bottom. at least after this check: ```if (message.data.name !== 'bewotecDataTransfer') {```

instead you would close the bridge window at any "non-bridge"-message event too.

and can you please add a log message, that you are now closing the bridge window.

Reply:
do not forget a test for that